src/fheroes2/castle/castle_mageguild.cpp

```diff
@@ -112,18 +127,48 @@ void Castle::_openMageGuild( const Heroes * hero ) const
 
         spellRows[levelIndex] = std::make_unique<fheroes2::SpellsInOneRow>( std::move( spells ) );
 
-        spellRows[levelIndex]->setPosition( { cur_pt.x + 250, cur_pt.y + 365 - 90 * static_cast<int32_t>( levelIndex ) } );
+        spellRows[levelIndex]->setPosition( { dialogOffset.x + 250, dialogOffset.y + 365 - 90 * static_cast<int32_t>( levelIndex ) } );
         spellRows[levelIndex]->redraw( display );
     }
 
-    fheroes2::Button buttonExit( cur_pt.x + fheroes2::Display::DEFAULT_WIDTH - exitWidth, cur_pt.y + bottomBarOffsetY, ICN::BUTTON_GUILDWELL_EXIT, 0, 1 );
+    if ( GetKingdom().GetCastles().size() < 2 ) {
+        buttonPrevCastle.disable();
+        buttonNextCastle.disable();
+    }
 
-    buttonExit.draw();
+    buttonPrevCastle.draw();
+    buttonNextCastle.draw();
 
-    display.render();
+    buttonExit.draw();
 
     LocalEvent & le = LocalEvent::Get();
 
+    auto updateStatusBar = [&le, &buttonExit, &buttonPrevCastle, &buttonNextCastle, &statusBar, &defaultStatusBarText]() {
+        std::string statusMessage;
+
+        if ( le.isMouseCursorPosInArea( buttonExit.area() ) ) {
+            statusMessage = _( "Exit Mage Guild" );
+        }
+        else if ( buttonPrevCastle.isEnabled() && le.isMouseCursorPosInArea( buttonPrevCastle.area() ) ) {
+            statusMessage = _( "Show previous town" );
+        }
+        else if ( buttonNextCastle.isEnabled() && le.isMouseCursorPosInArea( buttonNextCastle.area() ) ) {
+            statusMessage = _( "Show next town" );
+        }
+
+        if ( statusMessage.empty() ) {
+            statusBar.ShowMessage( defaultStatusBarText );
+        }
+        else {
+            statusBar.ShowMessage( statusMessage );
+            statusMessage.clear();
+        }
+    };
+
+    updateStatusBar();
+
+    display.render();
```

Comment:
We can slightly optimize the rendering:

```suggestion
    display.render( restorer.rect() );
```